components/admin_console/s3_connection_test/s3_connection_test.jsx

```diff
@@ -0,0 +1,128 @@
+
+        const config = JSON.parse(JSON.stringify(this.props.config));
+        this.props.getConfigFromState(config);
+
+        this.props.actions.testS3Connection(config).then(
+            (data) => {
+                if (data.error) {
+                    let fail = data.error.message;
+                    if (data.error.detailed_error) {
+                        fail += ' - ' + data.error.detailed_error;
+                    }
+                    this.setState({
+                        testing: false,
+                        fail
+                    });
+                } else {
+                    this.setState({
+                        testing: false,
+                        success: true
+                    });
+                }
+            }
+        );
+    }
+
+    render() {
+        let testMessage = null;
+        if (this.state.success) {
+            testMessage = (
+                <div className='alert alert-success'>
+                    <i className='fa fa-check'/>
+                    <FormattedMessage
+                        id='admin.s3.s3Success'
+                        defaultMessage='Connection was successful: No errors were reported while testing the S3 access.'
+                    />
+                </div>
+            );
+        } else if (this.state.fail) {
+            testMessage = (
+                <div className='alert alert-warning'>
+                    <i className='fa fa-warning'/>
+                    <FormattedMessage
+                        id='admin.s3.s3Fail'
+                        defaultMessage='Connection unsuccessful: {error}'
+                        values={{
+                            error: this.state.fail
+                        }}
+                    />
+                </div>
+            );
+        }
+
+        let contents = null;
+        if (this.state.testing) {
+            contents = (
+                <span>
+                    <span className='fa fa-refresh icon--rotate'/>
+                    {Utils.localizeMessage('admin.s3.testing', 'Testing...')}
+                </span>
+            );
+        } else {
+            contents = (
+                <FormattedMessage
+                    id='admin.s3.connectionS3Test'
+                    defaultMessage='Test Connection'
+                />
+            );
+        }
+
+        return (
+            <div className='form-group email-connection-test'>
+                <div className='col-sm-offset-4 col-sm-8'>
+                    <div className='help-text'>
+                        <button
+                            className='btn btn-default'
+                            onClick={this.handleTestConnection}
+                            disabled={this.props.disabled}
+                        >
+                            {contents}
+                        </button>
+                        {testMessage}
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
```

Comment:
Please add component test, being new component :)

Reply:
I think the test you've made is fine. Thanks!

Reply:
thanks!